web/modules/custom/farm_nfa/js/farmOS.map.behaviors.farm_nfa_gfw_layers.js

```diff
@@ -2,13 +2,83 @@
     farmOS.map.behaviors.farm_nfa_gfw_layers = {
       attach: async function (instance) {
         // Add layers for fire and deforestation alerts in the GFW plan tab
-        farmNfaPlotGfwApiMap(instance,'fire', 'https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/v20220726/query/json');
-        farmNfaPlotGfwApiMap(instance,'deforestation', 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json');
+        const fireAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/v20220726/query/json';
+        const deforestationAlertsUrl = 'https://data-api.globalforestwatch.org/dataset/gfw_integrated_alerts/v20230215/query/json';
+        farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl);
+        farmNfaPlotGfwApiMap(instance,'deforestation', deforestationAlertsUrl);
+        const dateApplyButton = document.querySelector('.comiseo-daterangepicker-buttonpanel button');
+        const dateUpdateButtons = document.querySelectorAll('[role="menuitem"]');
+        dateUpdateButtons.forEach((dateUpdateButton) => {
+          dateUpdateButton.addEventListener('click', () => {
+            updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+          });
+        });
+        dateApplyButton.addEventListener('click', () => {
+          updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl);
+        });
       }
     }
 }())
 
-async function farmNfaPlotGfwApiMap(instance, mapType, gfwApiUrl) {
+async function updateMapLayers(instance, fireAlertsUrl, deforestationAlertsUrl) {
+  const dateRange = await getStartEndDate();
+  const map = instance.map;
+  const layers = map.getLayers().getArray();
+  const noOfLayers = layers.length;
+  for (let i = 2; i < noOfLayers; i++) {
+    await map.removeLayer(layers[layers.length - 1]);    
+  }
+  farmNfaPlotGfwApiMap(instance, 'fire', fireAlertsUrl, dateRange);
+  farmNfaPlotGfwApiMap(instance, 'deforestation', deforestationAlertsUrl, dateRange);
+}
+
+function getStartEndDate() {
+  return new Promise((resolve, reject) => { 
+    setTimeout(() => {
```

Comment:
A couple more things, we should avid timeouts and use request animation frame instead https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame

Any reason why are you using promises here?

Reply:
@pcambra the daterangepicker library was taking some time to update the UI, due to which while parsing that info the old selected dates were getting parsed, to solve this issue i added this timeouts so that we can parse the updated information from it !
the promises are used to await the script till we parse the correct dates, else it will move forward without getting the values and we will get `undefined` for the daterange

Reply:
Let's see if the daterangepicker library gives us events instead.